### ISSUE TITLE
Retraining and updating an existing Tensorflow model

### ISSUE THREAD

**Issue by manaschaturvedi** [RasaHQ/rasa_nlu#1082]
I've been using Rasa NLU for a project which involves making sense of structured text. My use case requires me to keep updating my training set by adding new examples of text corpus entities. However, this means that I have to keep retraining my model every few days, thereby taking more time for the same owing to increased training set size.

Is there a way in Rasa NLU to update an already trained model by only training it with the new training set data instead of retraining the entire model again using the entire previous training data set and the new training data set?

I'm trying to look for an approach where I can simply update my existing trained model by training it with incremental additional training data set every few days.

**Comment by amn41** [comment]
this should (in theory) be possible with the tensorflow intent classifier, but you would have to make sure the countvectorfeaturizer can also do this (e.g. not shuffle the word order). So in theory yes, but would probably be a larger PR to get this to work.

**Comment by manaschaturvedi** [comment]
Sorry I'm not quite sure if I get what you mean. Is there an inbuilt functionality in Rasa NLU to update an existing model with a new training data set without retraining it along with the previous training sets as well?

I'd really appreciate if you could guide me in this specific use case. Thanks.

**Comment by amn41** [comment]
no there is no such functionality at the moment. If you are interested in working on it, we would be happy to support you in creating a PR though!

**Comment by r4sn4** [comment]
@amn41 Can you guide what could be possible workaround for the same?

**Comment by amn41** [comment]
@Ghostvv is the best person to advise

**Comment by r4sn4** [comment]
@Ghostvv could you please tell any workaround for this?

**Comment by Ghostvv** [comment]
You would need to add quite a bit of custom code in order to support continue training. First of all, if you expect for the new words to appear, then you'd need to create fix vocabulary and hack into `intent_featurizer_count_vectors` to feed it to sklearns's CountVectorizer. Second, you'd need to persist `train_op` in `intent_classifier_tensorflow_embedding`. Finally add a method that would use persisted `tf.Session` and `train_op` to update the weights (you could use this as an inspiration: https://github.com/RasaHQ/rasa_core/blob/b362fdc9c609c7078350a32198310137caa8c49f/rasa_core/policies/embedding_policy.py#L722)

**Comment by imadcat** [comment]
When using EmbeddingPolicy, this is a must have feature, when you have a model trained on 1000 stories, and then need to add 10 new stories to update the model, you don't want to waste tens of hours to retrain on the 1000 old stories. For other policies, training is so fast, retraining on full dataset is not a problem.

**Comment by Ghostvv** [comment]
@imadcat embedding policy supports continue training, but only if it was trained on the same instance of the class, `continue_training` after loading is not supported

**Comment by rohitiec** [comment]
Hi @Ghostvv , I am using supervised_embeddings for training my data, How can i achieve the same? i did't find anything related to this on rasa docs.

**Comment by Ghostvv** [comment]
it is not supported, you'd need to modify it yourself. See above for a hint on how to do it

**Comment by DivyaWadehra** [comment]
is it working?
